livemaker/cli/lmpatch.py

```diff
@@ -118,25 +124,39 @@ def bar_show(item):
                 for info in bar:
-                    if info.path == lsb_path:
-                        # replace existing with patch version
-                        #
-                        # TODO: support writing encrypted files
-                        if info.compress_type == LMCompressType.ENCRYPTED:
-                            compress_type = LMCompressType.NONE
-                        elif info.compress_type == LMCompressType.ENCRYPTED_ZLIB:
-                            compress_type = LMCompressType.ZLIB
+                    # replace existing with patch version
+                    #
+                    # TODO: support writing encrypted files
+                    if info.compress_type == LMCompressType.ENCRYPTED:
+                        compress_type = LMCompressType.NONE
+                    elif info.compress_type == LMCompressType.ENCRYPTED_ZLIB:
+                        compress_type = LMCompressType.ZLIB
+                    else:
+                        compress_type = info.compress_type
+
+                    if path_mode:
+                        thisfile = Path(patched_lsb).joinpath(info.path)
+                        if os.path.exists(thisfile):
+                            new_lm.write_as(thisfile, info.path, compress_type=compress_type, unk1=info.unk1)
+                            log.info('patched {}'.format(info.path))
+                            # print('patched'.format(info.path))
                         else:
-                            compress_type = info.compress_type
-                        new_lm.write(lsb_path, compress_type=compress_type, unk1=info.unk1)
-                        log.info('patched {}'.format(lsb_path))
-                        # print('patched')
+                            # copy original version
+                            data = orig_lm.read(info, decompress=False)
+                            new_lm.writebytes(info, data)
+                            log.info('copied {}'.format(info.path))
+                            # print('copied {}'.format(info.path))
                     else:
-                        # copy original version
-                        data = orig_lm.read(info, decompress=False)
-                        new_lm.writebytes(info, data)
-                        log.info('copied {}'.format(lsb_path))
-                    # print(info.name)
+                        if info.path == lsb_path:
+                            new_lm.write(lsb_path, compress_type=compress_type, unk1=info.unk1)
+                            log.info('patched {}'.format(lsb_path))
+                            # print('patched')
+                        else:
+                            # copy original version
+                            data = orig_lm.read(info, decompress=False)
+                            new_lm.writebytes(info, data)
+                            log.info('copied {}'.format(lsb_path))
+                            # print(info.name)
```

Comment:
all of this duplicated code should be cleaned up

something along the lines of:

```py
arcname=None
if recursive:
    lsb_path = Path(patched_lsb).joinpath(info.path)
    if lsb_path.exists():
        arcname=info.path
    else:
        lsb_path = None
else:
    lsb_path = PureWindowsPath(patched_lsb)
    if info.path != lsb_path:
        lsb_path = None

if lsb_path:
    # patch
    new_lm.write(lsb_path, arcname=arcname, ...)
    ...
else:
    # copy original
    new_lm.writebytes(...)
    ...
```

Reply:
OK